types/ws/index.d.ts

```diff
@@ -230,15 +234,18 @@ declare namespace WebSocket {
         on(event: 'connection', cb: (this: WebSocket, socket: WebSocket, request: http.IncomingMessage) => void): this;
         on(event: 'error', cb: (this: WebSocket, error: Error) => void): this;
         on(event: 'headers', cb: (this: WebSocket, headers: string[], request: http.IncomingMessage) => void): this;
-        on(event: 'listening', cb: (this: WebSocket) => void): this;
+        on(event: 'closed' | 'listening', cb: (this: WebSocket) => void): this;
```

Comment:
I think the event name is actually `close`. https://github.com/websockets/ws/blob/0a612364e69fc07624b8010c6873f7766743a8e3/doc/ws.md#event-close